Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -375,6 +376,18 @@ protected virtual Page CreatePageFromSegment(string segment)
             }
         }
 
+        protected void ApplyPageBehaviors(Page page)
```

Comment:
Just making it virtual would allow those who want to get crazy to implement a more complex scenario like the one you describe. Then `IPageBehaviorFactory` could be injected into the derived navigation service implementation and thus leave the default implementation free from those details. 

That is if there isn't a good reason to implement the `IPageBehaviorFactory`. Perhaps allows to handle more complex situations that will arise in the future without needing to change `NavigationService` and if so, this might be the good reason :). But what do I know

I haven't used that feature in WPF so can't tell if it's a good match for Forms. Probably since it's a good enough for WPF :) (if not crazy large)